test/Datadog.Trace.IntegrationTests/OriginTagSendTraces.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using MsgPack;
+using Xunit;
+
+namespace Datadog.Trace.IntegrationTests
+{
+    public class OriginTagSendTraces
+    {
+        private readonly Tracer _tracer;
+        private readonly TestApi _testApi;
+
+        public OriginTagSendTraces()
+        {
+            var settings = new TracerSettings();
+            _testApi = new TestApi();
+            var agentWriter = new AgentWriter(_testApi, statsd: null);
+            _tracer = new Tracer(settings, agentWriter, sampler: null, scopeManager: null, statsd: null);
+        }
+
+        [Fact]
+        public void NormalSpan()
+        {
+            var scope = _tracer.StartActive("Operation");
+            scope.Dispose();
+
+            var objects = _testApi.Wait();
```

Comment:
Probably just me, but I feel like the locking around the TestApi makes it seem more complicated than it is. Couldn't we just use `await FlushTracesAsync()` and do away with the `ManualResetEventSlim` in the test API for simplicity? Not suggesting you _should_ do that, more if there's a reason you didn't?

Reply:
the problem with `FlushTracesAsync` is that can return before the complete flush (can be caused by a previous flush from the timer/delay loop), so can introduce a race condition. That's the reason I had to add the 2 seconds delay in the PR of the testing framework test suite. I think @kevingosse is going to handle this on the agent writer later or we can add it to the backlog, depending on his thoughts about this.